client/client.go

```diff
@@ -104,6 +104,10 @@ func getAccessTokenFromJSON(body []byte) (accessToken string, err error) {
 		err = errors.WithMessage(err, "unable to unmarshall the accessToken, "+string(body))
 		return
 	}
+	if accessTokenStruct.Message == "Unauthorized access" {
```

Comment:
I think the message returned by the server has "Unauthorized access" as a substring, but the full message is longer.

Would it be better to check the http response status code? It looks like the status code is checked in the function `exhangeRefreshTokenForAccessToken()` on line 84 above, but there's a bug. Line 85 sets the return param `err` but then the next line is missing `return`.

One other nit: line 85 is using `errors.New(fmt.Sprintf(...))` which can be simplified to `errors.Errorf(...)`

Reply:
Correction: the message is "Unauthorized access" if the "refresh token" includes < 2 "." (period) characters. Seems fragile to depend on that.

Reply:
+1

Reply:
@TheMattSchiller @yoshiotu i'd be more than happy to chat with you two about this project